scripts/plot.py

```diff
@@ -0,0 +1,147 @@
+#!/usr/bin/env python3
+"""plot.py
+
+Usage:
+  plot.py -a ALGORITHM -g GEOMETRY -i INPUT... [-d] [-l] [-y] [-n] [-o OUTPUT_FILE]
+  plot.py (-h | --help)
+
+Options:
+  -h --help                         Show this screen.
+  -a ALGORITHM --algo ALGORITHM     Algorithm ['construction', 'knn_search', 'radius_search', 'scaling_table']
+  -g GEOMETRY --geometry GEOMETRY   Geometry (source cloud) ['filled_box', 'hollow_box']
+  -i FILE --input-files=FILE        Input file containing benchmark results
+  -o FILE --output-file=FILE        Output file [default: plot.png]
+  -p PLOT_INDEX --plot=PLOT_INDEX   Plot index
+  -d --display                      Display mode
+  -l --legend                       Add legend
+  -y --ylabel                       Add y-axis label
+  -n --numbers                      Print numbers
+"""
+import matplotlib.pyplot as plt
+import numpy as np
+import math
+import re
+from docopt import docopt
+
+from tableau import tableau20
+
+from benchmark import parse_benchmark
+
+def libraries_comparison_rate_figure(input_file, algorithm, geometry, numbers):
+    # Phase 0: parse backends
+    backends = set()
+    with open(input_file) as f:
+        for line in f:
+            search_result = re.search("[^<]*<ArborX::BVH<([^>]*)>.*", line)
+            if search_result != None:
+                backends.add(search_result.group(1))
+    backends = list(backends)
+    backends.sort()
```

Comment:
I like the automatic way. I also think it should have a manual mode. For example, you may want to get a subset of backends to plot.